spark-operator/src/main/java/org/apache/spark/k8s/operator/utils/PodPhase.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.spark.k8s.operator.utils;
+
+import io.fabric8.kubernetes.api.model.Pod;
+
+public enum PodPhase {
+  PENDING("pending"),
+  RUNNING("running"),
+  FAILED("failed"),
+  SUCCEEDED("succeeded"),
+  TERMINATING("terminating"),
+  UNKNOWN("unknown");
+
+  private final String description;
+
+  PodPhase(String description) {
+    this.description = description;
+  }
+
+  public static PodPhase getPhase(final Pod pod) {
+    if (pod != null && pod.getStatus() != null) {
```

Comment:
Please use a short-cut instead of main code nesting.

For example, at the beginning, 

```java
if (pod == null || pod.getStatus() == null) {
  return UNKNOWN;
}
```